packages/indexer-agent/src/network.ts

```diff
@@ -112,16 +124,123 @@ export class Network {
+          attempts: txConfig.attempt,
+        })
+        break
+      }
+
+      try {
+        if (txConfig.attempt > 1) {
+          logger.info('Resubmitting transaction', {
+            txConfig: txConfig,
+          })
+          const transaction: providers.TransactionRequest = {
+            value: tx.value,
+            to: tx.to,
+            data: tx.data,
+            chainId: tx.chainId,
+            from: tx.from,
+            nonce: txConfig.nonce,
+            gasPrice: Math.ceil(txConfig.gasPrice),
+            gasLimit: Math.ceil(txConfig.gasLimit),
+          }
+          tx = await this.wallet.sendTransaction(transaction)
+        }
+
+        logger.info(`Transaction pending`, { tx: tx })
+
+        const receipt = await this.ethereum.waitForTransaction(
+          tx?.hash,
+          3,
+          this.gasBumpTimeLimit,
+        )
+
+        if (receipt.status == 0) {
+          const tx = await this.ethereum.getTransaction(receipt.transactionHash)
+          const code = await this.ethereum.call(tx)
+          throw Error(`Transaction reverted, reason: ${code}`)
+        }
+
+        logger.info(`Transaction successfully included in block`, {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          tx: tx?.hash,
+          receipt: receipt,
+        })
+        output = receipt
+        pending = false
+      } catch (error) {
+        txConfig = await this.updateTransactionConfig(logger, txConfig, error)
+        continue
+      }
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return output!
+  }
+
+  async updateTransactionConfig(
+    logger: Logger,
+    txConfig: Record<string, number>,
+    error: Error,
+  ): Promise<Record<string, number>> {
+    if (
+      error.message.includes(
+        'Transaction with the same hash was already imported',
+      )
+    ) {
+      txConfig.nonceOffset += 1
+      txConfig.nonce =
+        (await this.wallet.getTransactionCount('pending')) -
+        txConfig.nonceOffset +
+        1
+    } else if (
+      error.message.includes(
+        'Transaction nonce is too low. Try incrementing the nonce.',
+      )
+    ) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      txConfig.nonce! += 1
+    } else if (error.message.includes('Reverted')) {
+      txConfig.gasLimit = txConfig.gasLimit * txConfig.gasBump
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      txConfig.nonce! += 1
+    } else if (
+      error.message.includes('Try increasing the fee') ||
+      error.message.includes(
+        'Transaction gas price supplied is too low. There is another transaction with same nonce in the queue. Try increasing the gas price or incrementing the nonce.',
+      ) ||
+      error.message?.includes('timeout exceeded')
+    ) {
+      txConfig.gasPrice =
+        txConfig.gasPrice * txConfig.gasBump > this.gasPriceMax
```

Comment:
Maybe we can test if txConfig.gasPrice === txConfig.gasPriceMax then avoid resending as it create the same "try increasing fee" exception.